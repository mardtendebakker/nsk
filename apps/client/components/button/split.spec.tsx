import { fireEvent, render } from '@testing-library/react';
import Split from './split';

describe('split', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Split onClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('handles click', () => {
    const onClick = jest.fn();
    const { getByRole } = render(<Split onClick={onClick} />);
    fireEvent.click(getByRole('button'));

    expect(onClick).toHaveBeenCalled();
  });
  it('disables click', () => {
    const onClick = jest.fn();
    const { getByRole } = render(<Split onClick={onClick} disabled />);
    fireEvent.click(getByRole('button'));

    expect(onClick).not.toHaveBeenCalled();
  });
});
