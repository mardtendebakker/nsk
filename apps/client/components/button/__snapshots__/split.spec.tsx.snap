// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`split matches snapshot 1`] = `
<DocumentFragment>
  <span
    class=""
    data-mui-internal-clone-element="true"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1rb4sk0-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lr4af1-MuiSvgIcon-root"
        data-testid="BoltIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 21h-1l1-7H7.5c-.58 0-.57-.32-.38-.66.19-.34.05-.08.07-.12C8.48 10.94 10.42 7.54 13 3h1l-1 7h3.5c.49 0 .56.33.47.51l-.07.15C12.96 17.55 11 21 11 21"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </span>
</DocumentFragment>
`;
